miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppInfoFetcher.kt

```diff
@@ -5,17 +5,11 @@ import com.rakuten.tech.mobile.miniapp.api.UpdatableApiClient
 
 internal class MiniAppInfoFetcher(private var apiClient: ApiClient) : UpdatableApiClient {
 
+    @Throws(MiniAppSdkException::class)
     suspend fun fetchMiniAppList() = apiClient.list()
 
-    @Suppress("TooGenericExceptionCaught", "SwallowedException")
-    suspend fun getInfo(appId: String) = run {
-        try {
-            apiClient.fetchInfo(appId)
-        } catch (error: Exception) {
```

Comment:
I think catching specific `NoSuchElementException` might suffice. What do you think?

Reply:
Do you mean catching `NoSuchElementException` in `getInfo` or in `apiClient`?

Reply:
in `getInfo`

Reply:
Oh, in the case that it can't return `first` item? Hmm, maybe that should be handled in the ApiClient instead - we can check if the list is empty and return a `MiniAppSdkException` if so.

Reply:
Yeah for the first item. Until now, the handling/intelligence was in `MiniAppInfoFetcher` which was inline to `MiniAppDownloader` where manifest is also checked for. It's a very boundary-types one.

Reply:
I have updated this. I think in this case it makes more sense to check in `ApiClient` because we can directly check if the list is empty in that class.